pypeit/scienceimage.py

```diff
@@ -185,39 +188,60 @@ def proc_sci(self, file_list, reject_cr=True, sigma_clip=False, sigrej=None, max
               crmask
 
         """
-        nsci = len(file_list)
-        weights = np.ones(nsci)/float(nsci)
-        sciimg_stack, sciivar_stack, rn2img_stack, crmask_stack, mask_stack = \
-        self.read_stack(file_list, self.bias, self.pixel_flat, self.bpm, self.det, self.par['process'], self.spectrograph,
-                            illum_flat=self.illum_flat, reject_cr=reject_cr, binning=self.binning)
+        # Init
+        if ltype == 'sci':
+            combinedImage = self.sci_combine
+        elif ltype == 'bkg':
+            combinedImage = self.bkg_combine
+        else:
+            msgs.error("Bad ltype for proc_list")
+        nimg = combinedImage.nimages
+        weights = np.ones(nimg)/float(nimg)
 
-        # ToDO The bitmask is not being properly propagated here!
+        # Load
+        #embed(header='198 of sciimg')
+        combinedImage.load_images()
```

Comment:
See my other comments on this. I seen no reason to load in all the raw frames and then process them one by on. Just load them in one at a time, process them, and the raw frames can cease to exist saving memory. It makes more sense to me that the interface for processing a single image (trim, flat fielding, bias subtraction, overscan subtraction) takes place at the single image image level. Then the raw frame can cease to exist saving memory.

Reply:
I estimate 10 double-precision 4kx4k images is 1Gb of RAM.

Hard to see how this is a `massive memory hog`.  

All of the extras that are needed to `build_stack` are flushed
on the fly.